### PR TITLE
Add value throttled to DiscreteSlider

### PR DESCRIPTION
**Pull request by Hoxbro**
Supersede the bug fixes and documentation update in #1400

There has been some changes to the codebase since the previously PR, most notably it seems like #1256 has been fixed. 

So this PR updates the DiscreteSlider with the ability to use value_throttled as an argument #1253. I could not get value_throttled to work with the `constant=True`, which is why it is disabled. The PR also updates unit test for sliders with `value_throttled`.

Lastly the sliders examples/documentation with value_throttled has been updated #1255.

### CODE REVIEW

**Comment by Hoxbro** [comment]
@philippjfr do you have any comments or suggestion? And do you have a way to get `value_throttled = param.Parameter(constant=True)` to work if it is needed.

File: panel/widgets/slider.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ class DiscreteSlider(CompositeWidget, _SliderBase):
 
     value = param.Parameter()
 
-    value_throttled = param.Parameter(constant=True)
+    value_throttled = param.Parameter()
```
Comment (philippjfr) [inline panel/widgets/slider.py:178]: Setting these to constant was intentional. Users should generally only set `value` and only use `value_throttled` to watch it. So use the `edit_constant` property to change it internally.

**Comment by philippjfr** [comment]
I had one comment about keeping `value_throttled` as constant, otherwise this looks good to me. Thanks for reviving this work!

**Comment by matsvandecavey** [comment]
@philippjfr With the `value_throttled` as constant, is there a way to change its value programmatically?
My app has a *reset* button, but I can't change a constant value:
```python
myslider.value_throttled = default_value
```
This throws a `TypeError` 
> TypeError: Constant parameter 'value_throttled' cannot be modified

**Comment by philippjfr** [comment]
@matsvandecavey Change the `.value` and `.value_throttled` will update.

**Comment by matsvandecavey** [comment]
Hi Philipp, This is not working for me.

value_throttled

**Pull request merged**